src/api/Feed.js

```diff
@@ -182,94 +170,66 @@ class Feed extends Base {
      * Fetches the comments for a file
      *
      * @param {Object} permissions - the file permissions
-     * @param {Function} errorCallback - the function which will be called on error
      * @return {Promise} - the file comments
      */
-    fetchComments(permissions: BoxItemPermission, errorCallback: ErrorCallback): Promise<?Comments> {
+    fetchComments(permissions: BoxItemPermission): Promise<?Comments> {
         this.commentsAPI = new CommentsAPI(this.options);
         return new Promise(resolve => {
             this.commentsAPI.getComments(
                 this.id,
                 permissions,
                 resolve,
-                (e: ElementsXhrError) => {
-                    this.fetchFeedItemErrorCallback(e, errorCallback, ERROR_CODE_FETCH_COMMENTS);
-                    resolve();
-                },
-                DEFAULT_START,
-                DEFAULT_END,
-                COMMENTS_FIELDS_TO_FETCH,
-                true,
+                this.fetchFeedItemErrorCallback.bind(this, resolve),
             );
         });
     }
 
     /**
      * Fetches the versions for a file
      *
-     * @param {Function} errorCallback - the function which will be called on error
      * @return {Promise} - the file versions
      */
-    fetchVersions(errorCallback: ErrorCallback): Promise<?FileVersions> {
+    fetchVersions(): Promise<?FileVersions> {
         this.versionsAPI = new VersionsAPI(this.options);
         return new Promise(resolve => {
-            this.versionsAPI.offsetGet(
-                this.id,
-                resolve,
-                (e: ElementsXhrError) => {
-                    this.fetchFeedItemErrorCallback(e, errorCallback, ERROR_CODE_FETCH_VERSIONS);
-                    resolve();
-                },
-                DEFAULT_START,
-                DEFAULT_END,
-                VERSIONS_FIELDS_TO_FETCH,
-                true,
-            );
+            this.versionsAPI.getVersions(this.id, resolve, this.fetchFeedItemErrorCallback.bind(this, resolve));
```

Comment:
It seems odd that the inner methods would resolve successfully after an error. Ideally, they would reject and the parent should catch and handle them appropriately.

Reply:
Agreed. The reason why they resolve undefined is that `Promise.all` is being used to fetch the comments, tasks, versions. If we reject, we cannot show partial data as it will hit the .catch.

A custom promise.all could be implemented (or imported) to handle this. Another unfortunate thing about reject is that it can only return 1 value, but I need 2 (error and code from the error callback). We have an eslint rule that enforces it to be of type Error. So, I could create an Error object with the code stuck into it.